src/Dodo.HttpClient.ResiliencePolicies/RetryPolicy/RetryPolicySettings.cs

```diff
@@ -1,28 +1,116 @@
 using Polly;
+using Polly.Contrib.WaitAndRetry;
 
 namespace Dodo.HttpClientResiliencePolicies.RetryPolicy
 {
 	public class RetryPolicySettings : IRetryPolicySettings
 	{
-		public ISleepDurationProvider SleepDurationProvider { get; }
+		private readonly IEnumerable<TimeSpan> _sleepDurationProvider;
+		IEnumerable<TimeSpan> IRetryPolicySettings.SleepDurationProvider => _sleepDurationProvider;
+
 		public Action<DelegateResult<HttpResponseMessage>, TimeSpan> OnRetry { get; set; }
 
 		public RetryPolicySettings()
-		: this(RetryPolicy.SleepDurationProvider.Jitter(
-			Defaults.Retry.RetryCount,
-			TimeSpan.FromMilliseconds(Defaults.Retry.MedianFirstRetryDelayInMilliseconds)))
 		{
+			_sleepDurationProvider = SleepDurationProvider.Jitter(
+				Defaults.Retry.RetryCount,
+				TimeSpan.FromMilliseconds(Defaults.Retry.MedianFirstRetryDelayInMilliseconds));
+
+			OnRetry = DoNothingOnRetry;
 		}
 
-		public RetryPolicySettings(
-			ISleepDurationProvider sleepDurationProvider)
+		private RetryPolicySettings(
+			IEnumerable<TimeSpan> sleepDurationProvider)
 		{
-			SleepDurationProvider = sleepDurationProvider;
+			if (sleepDurationProvider == null)
+				throw new ArgumentNullException(nameof(sleepDurationProvider));
+
+			_sleepDurationProvider = sleepDurationProvider;
 			OnRetry = DoNothingOnRetry;
 		}
 
 		private static readonly Action<DelegateResult<HttpResponseMessage>, TimeSpan> DoNothingOnRetry = (_, __) => { };
+
+		public static RetryPolicySettings Constant(int retryCount)
+		{
+			return Constant(retryCount, TimeSpan.FromMilliseconds(Defaults.Retry.InitialDelayMilliseconds));
+		}
+
+		public static RetryPolicySettings Constant(int retryCount, TimeSpan delay)
+		{
+			return new RetryPolicySettings(SleepDurationProvider.Constant(retryCount, delay));
+		}
+
+		public static RetryPolicySettings Linear(int retryCount)
+		{
+			return Linear(retryCount, TimeSpan.FromMilliseconds(Defaults.Retry.InitialDelayMilliseconds));
+		}
+
+		public static RetryPolicySettings Linear(int retryCount, TimeSpan initialDelay)
+		{
+			return new RetryPolicySettings(SleepDurationProvider.Constant(retryCount, initialDelay));
+		}
+
+		public static RetryPolicySettings Exponential(int retryCount)
+		{
+			return Exponential(retryCount, TimeSpan.FromMilliseconds(Defaults.Retry.InitialDelayMilliseconds));
+		}
+
+		public static RetryPolicySettings Exponential(int retryCount, TimeSpan initialDelay)
+		{
+			return new RetryPolicySettings(SleepDurationProvider.Exponential(retryCount, initialDelay));
+		}
+
+		public static RetryPolicySettings Jitter(int retryCount)
+		{
+			return Jitter(retryCount, TimeSpan.FromMilliseconds(Defaults.Retry.MedianFirstRetryDelayInMilliseconds));
+		}
+
+		public static RetryPolicySettings Jitter(int retryCount, TimeSpan medianFirstRetryDelay)
+		{
+			return new RetryPolicySettings(SleepDurationProvider.Jitter(retryCount, medianFirstRetryDelay));
+		}
+
+		#region nested class
+
+		private static class SleepDurationProvider
```

Comment:
Should we leave it here or move to separate file for readability purpose?

Reply:
I don't have a strong opinion about it... Personally, I don't like the nested classes at all. But! In this case, the one reason why I've had to make it is I can't use language ability to overload static methods and non-static methods. The static one I need to use it in the constructor, non-static - outside the class.